orchagent/policerorch.cpp

```diff
@@ -123,134 +141,403 @@ void PolicerOrch::doTask(Consumer &consumer)
+            {
+                attr.id = SAI_POLICER_ATTR_CBS;
+                attr.value.u64 = stoul(value);
+            }
+            else if (field == cir_field)
+            {
+                attr.id = SAI_POLICER_ATTR_CIR;
+                attr.value.u64 = stoul(value);
+            }
+            else if (field == pbs_field)
+            {
+                attr.id = SAI_POLICER_ATTR_PBS;
+                attr.value.u64 = stoul(value);
+            }
+            else if (field == pir_field)
+            {
+                attr.id = SAI_POLICER_ATTR_PIR;
+                attr.value.u64 = stoul(value);
+            }
+            else if (field == red_packet_action_field)
+            {
+                attr.id = SAI_POLICER_ATTR_RED_PACKET_ACTION;
+                attr.value.s32 = packet_action_map.at(value);
+            }
+            else if (field == green_packet_action_field)
+            {
+                attr.id = SAI_POLICER_ATTR_GREEN_PACKET_ACTION;
+                attr.value.s32 = packet_action_map.at(value);
+            }
+            else if (field == yellow_packet_action_field)
+            {
+                attr.id = SAI_POLICER_ATTR_YELLOW_PACKET_ACTION;
+                attr.value.s32 = packet_action_map.at(value);
+            }
+            else
+            {
+                SWSS_LOG_ERROR("Unknown policer attribute %s specified",
+                        field.c_str());
+                continue;
+            }
 
-                SWSS_LOG_DEBUG("attribute: %s value: %s", field.c_str(), value.c_str());
+            attrs.push_back(attr);
+        }
 
-                sai_attribute_t attr;
+        // Create a new policer
+        if (!update)
+        {
+            if (!meter_type || !mode)
+            {
+                SWSS_LOG_ERROR("Failed to create policer %s,\
+                        missing mandatory fields", key.c_str());
+                return task_process_status::task_invalid_entry;
+            }
 
-                if (field == meter_type_field)
-                {
-                    attr.id = SAI_POLICER_ATTR_METER_TYPE;
-                    attr.value.s32 = (sai_meter_type_t) meter_type_map.at(value);
-                    meter_type = true;
-                }
-                else if (field == mode_field)
-                {
-                    attr.id = SAI_POLICER_ATTR_MODE;
-                    attr.value.s32 = (sai_policer_mode_t) policer_mode_map.at(value);
-                    mode = true;
-                }
-                else if (field == color_source_field)
+            sai_object_id_t policer_id;
+            sai_status_t status = sai_policer_api->create_policer(
+                    &policer_id, gSwitchId, (uint32_t)attrs.size(), attrs.data());
+            if (status != SAI_STATUS_SUCCESS)
+            {
+                SWSS_LOG_ERROR("Failed to create policer %s, rv:%d",
+                        key.c_str(), status);
+                return task_process_status::task_need_retry;
```

Comment:
When do we need this retry? are we doing this to handle any dependent config's sequencing issues?

Reply:
Originally the PolicerOrch was implemented as a doTask which runs as a while loop and increments the iterator when create_policer fails. 
After splitting the handlers as handlePortStormControlTable and handlePolicerTable, the same logic is retained by returning task_process_status::task_need_retry. doTask increments the iterator when the handlers return task_need_retry. 
If we return failure, the iterator gets removed. 

w.r.t config dependency, the policers are created independently without any config dependency.